includes/data-port/models/class-sensei-import-model.php

```diff
@@ -0,0 +1,240 @@
+				continue;
+			}
+
+			$config = $schema_array[ $key ];
+			$value  = trim( $value );
+
+			if ( null !== $value ) {
+				switch ( $config['type'] ) {
+					case 'int':
+						$value = intval( $value );
+						break;
+					case 'float':
+						$value = floatval( $value );
+						break;
+					case 'bool':
+						$value = boolval( $value );
+						break;
+					case 'slug':
+						$value = sanitize_title( $value );
+						break;
+					case 'email':
+						$value = sanitize_email( $value );
+						break;
+					case 'url-or-file':
+						$value = 0 === strpos( $value, 'http' ) ? esc_url_raw( $value ) : sanitize_file_name( $value );
+						break;
+					case 'username':
+						$value = sanitize_user( $value );
+						break;
+					case 'video':
+						$value = Sensei_Wp_Kses::maybe_sanitize( $value, Sensei_Course::$allowed_html );
+						break;
+					default:
+						if (
+							isset( $config['pattern'] )
+							&& 1 !== preg_match( $config['pattern'], $value )
+						) {
+							$value = null;
+						} elseif ( ! empty( $config['allow_html'] ) ) {
+							$value = trim( wp_kses_post( $value ) );
+						} else {
+							$value = sanitize_text_field( $value );
+						}
+				}
+			}
+
+			$sanitized_data[ $key ] = $value;
+		}
+
+		$this->set_data( $sanitized_data );
+	}
+
+
+	/**
+	 * Adds a thumbnail to a post. The source of the thumbnail can be either a filename from the media library or an
+	 * external URL.
+	 *
+	 * @param string $column_name  The CSV column name which has the image source.
+	 *
+	 * @return bool|WP_Error  True on success, WP_Error on failure.
+	 */
+	protected function add_thumbnail_to_post( $column_name ) {
+		$post_id   = $this->get_post_id();
+		$thumbnail = $this->get_value( $column_name );
+
+		if ( null === $thumbnail ) {
+			return true;
+		}
+
+		if ( '' === $thumbnail ) {
+			delete_post_meta( $post_id, '_thumbnail_id' );
+		} else {
+			$attachment_id = Sensei_Data_Port_Utilities::get_attachment_from_source( $thumbnail );
+
+			if ( is_wp_error( $attachment_id ) ) {
+				return $attachment_id;
+			}
+
+			update_post_meta( $post_id, '_thumbnail_id', $attachment_id );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Stores an import id to the job.
+	 */
+	protected function store_import_id() {
+		$import_id = $this->get_value( $this->schema->get_column_id() );
+
+		if ( ! empty( $import_id ) ) {
+			$this->import_job->set_import_id( $this->schema->get_post_type(), $import_id, $this->get_post_id() );
+		}
+	}
+
+	/**
+	 * Returns the post id for an import id or check if the post exists.
+	 *
+	 * @param string $post_type  The post type.
+	 * @param string $import_id  The import id.
+	 *
+	 * @return int|null The post id if the post exists, null otherwise.
+	 */
+	protected function translate_import_id( $post_type, $import_id ) {
+		if ( empty( $import_id ) ) {
+			return null;
+		}
+
+		if ( 0 === strpos( $import_id, 'id:' ) ) {
+			return $this->import_job->get_import_id( $post_type, substr( $import_id, 3 ) );
```

Comment:
We should cast to `int`.
```suggestion
			return $this->import_job->get_import_id( $post_type, (int) substr( $import_id, 3 ) );
```

Reply:
As discussed: c3d9fd80d03ce0ba50fa327e5fa14ee4c28db7a9